### PR TITLE
`

### PR DESCRIPTION
**Pull request by live627**

### CODE REVIEW

**Comment by StealthWombat** [comment]
Isn't https://github.com/MustafaCan/SMF2.1/commit/9ef37f8d7e6fec1248ac5b912277073603836671 going to mean all content automatically goes to 100% width on any screen?

Or am I missing something? Offhand I can't see how you'd get any control without those divs.

I realise the main body of the forum can be controlled by .wrapper, but what about the content in the top bar, header, and footer? The .frame class was to keep those at the same overall width as the main body of the forum.

The easy way of dealing with responsive is to use max-width and min-width. These will override width, even if it's set as an inline style via the old admin setting.

Not that I think the old admin setting should be kept, but I do think some method of controlling width relative to screen width should be kept.

What I usually do is set a max-width in em\* in the CSS to stop things getting doofus wide on big screens, and then deal with responsive by just changing the max-width in the media queries.

Short version: having a width setting, either in admin or in index.css, is no barrier to responsive.

*I prefer using em here because if someone needs bigger text, they'll probably want more line length for it too.

**Comment by live627** [comment]
Maybe that specific commit broke things, but the complete branch still retains that 90% width we all know and love.

**Comment by StealthWombat** [comment]
Actually I don't love it and would happily ditch it. I just wouldn't necessarily ditch all the associated divs. ;)

**Comment by StealthWombat** [comment]
Aha. Had a look live on local and I see how he's done it. It works, but TBH I think it still might benefit from a common class, to allow easy setting of overall width without editing a pile of widely separated elements in the CSS. I'm thinking making it less aggravating for the average mug, if they want to alter it.

It's usual for custom themes to set overall width in the CSS, but to have it controlled in one spot. Usually near the top of the file, somewhere after the body tag declaration. It could still be down the file with all the #header etc declarations, but if so it'd be a good idea to provide a bookmark (like /\* Search #wrapper to set overall width */ or something) at the top of the file (which is not a bad idea for several things anyway).

If you need the short version, I'm thinking the common class may involve putting back one or two divs. Will have a bit more of a think about it.

**Comment by StealthWombat** [comment]
Ok, here's what I think would be a good idea. No need to reinstate all the divs that were taken out, but put back two of them: one in the top bar and one down in the footer.

Keep the existing #top_section, #header and #footer tags, but change the id of the current #wrapper to #forum just to make it more obvious.

Use the common class .wrapper to control the width of all sections. This will make things easy for people, while still being cleaner than the old setup. Pic attached.

Markup roughly:

```
<body>
    <div id="top_section">
        <div class="wrapper">
            User menu, search bar, etc
        </div>
    </div>
    <div id="header" class="wrapper">
            Forum title, banner, etc
    </div>
    <div id="forum" class="wrapper">
            All the forum stuff
    </div>
    <div id="footer">
        <div class="wrapper">
            Copyright, help, rules, etc
        </div>
    </div>
</body>
```

index_markup_recommendation
